### PR TITLE
Fix dropdown navigation #382

### PR DESCRIPTION
**Pull request by Dunkat**

### CODE REVIEW

File: src/components/widget/List/RawList.js [hunk 1]
```diff
@@ -60,18 +79,24 @@ class RawList extends Component {
...
-        const {selected} = this.state;
-        const {list} = this.props;
+        const {selected, dropdownList} = this.state;
+
+        let selectedIndex = null;
 
-        const next = up ? selected + 1 : selected - 1;
+        dropdownList.map((item, index) => {
+            if(JSON.stringify(item) === JSON.stringify(selected)){
+                return selectedIndex = index;
```
Comment (damianprzygodzki) [inline src/components/widget/List/RawList.js]: What you wanted to do here? It looks no good.

File: src/components/widget/List/RawList.js [hunk 2]
```diff
@@ -8,7 +8,26 @@ class RawList extends Component {
...
+
+        if(prevProps.list !== list){
+            let dropdown = [];
+
+            if(!mandatory){
+                dropdown.push(0);
+            }
+
+            if(list.length > 0) {
+                this.setState(Object.assign({}, this.state, {
```
Comment (damianprzygodzki) [inline src/components/widget/List/RawList.js]: @Dunkat We are going away from the assumption, to always provide deep copy of object, event if it is one level deep. `setState` is doing that for us.

File: src/components/widget/List/RawList.js (hunk 1 above)
Comment (Dunkat) [inline src/components/widget/List/RawList.js]: comparing two objects

Comment (Dunkat) [inline src/components/widget/List/RawList.js]: and return index of item which is selected to  have possibility get higher or lower index

Comment (damianprzygodzki) [inline src/components/widget/List/RawList.js]: So i believe the assignment `return selectedIndex = index;` is redundant.

File: src/components/widget/List/RawList.js [hunk 3]
```diff
@@ -60,18 +79,25 @@ class RawList extends Component {
...
-            selected: (next >= 0 && next <= list.length) ? next : selected
+        dropdownList.map((item, index) => {
+            if(JSON.stringify(item) === JSON.stringify(selected)){
+                selectedIndex = index;
+            }
         });
+
+        const next = up ? selectedIndex + 1 : selectedIndex - 1;
+
+        this.setState(Object.assign({}, this.state, {
```
Comment (damianprzygodzki) [inline src/components/widget/List/RawList.js]: The same thing, Object.assign to remove

**Pull request merged**